stock-tracker/src/features/keyStats/redux/actions.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  SET_RESPONSE,
+  SET_LOADING_KEYSTATS,
+  SET_ERROR_KEYSTATS
+} from "./actionTypes";
+import { Action, ActionWithPayload } from "../../../utils/actions";
+import * as constants from "./actionTypes";
+
+export interface ResponseData {
+  latestPrice: number;
+  change: number;
+  changePercent: number;
+  symbol: string;
+  companyName: string;
+  previousClose: number;
+  high: number | null;
+  low: number | null;
+  previousVolume: number;
+  marketCap: number;
+  peRatio: number;
+  open: number | null;
+  week52High: number;
+  week52Low: number;
+  avgTotalVolume: number;
+  earningsPerShare: number;
+  ytdChange: number;
+  currency: string | null;
+  latestTime: string;
+  latestUpdate: number;
+  isUSMarketOpen: boolean;
+}
+
+export type SetResponse = ActionWithPayload<
+  constants.SET_RESPONSE,
+  ResponseData
+>;
+
+export const setResponseAction = (response: ResponseData): SetResponse => ({
```

Comment:
You could go one step further to making this (and all other action creators) easier by making a `createAction` function.

You can make a very simple one, but you'll lose the `payload` type in that case. A more complex one would require some complex Typescript, though, so leave that for now.

Reply:
I have been looking at the Redux Starter Kit so might include this later on, but focusing on getting the whole project typed then refactor.